src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -372,9 +373,14 @@ private bool HasJustStartedFullSync(Snapshot best)
         private bool AnyDesiredPeerKnown(Snapshot best)
         {
             UInt256 localChainDifficulty = _syncProgressResolver.ChainDifficulty;
-            if (_logger.IsTrace) _logger.Trace($"Best peer [{best.PeerBlock},{best.PeerDifficulty}] > local [{best.Header},{localChainDifficulty}]");
-            return best.PeerDifficulty > localChainDifficulty ||
-                   best.PeerDifficulty == localChainDifficulty && best.PeerBlock > best.Header;
+            bool anyDesiredPeerKnown = best.PeerDifficulty > localChainDifficulty && best.HeaderHash != best.PeerHeadHash // we don't trust parity TotalDifficulty, so checking peerHeadHash. 
```

Comment:
easier to read if you keep ordering -> local always on left or right, now we have comparison remote > local && local != remote